src/modules/dashboard/components/ActionsPage/ActionsPage.tsx

```diff
@@ -299,6 +303,55 @@ const ActionsPage = () => {
     tokenInfo: { decimals, symbol },
   } = tokenData;
 
+  const getFormmatedRoleList = (roleGroupA, roleGroupB) => {
+    let roleList = '';
+
+    roleGroupA.forEach((role, i) => {
+      const roleNameMessage = { id: `role.${role.id}` };
+      const formattedRole = formatMessage(roleNameMessage);
+
+      roleList += ` ${formattedRole}`;
+
+      if (
+        i < roleGroupA.length - 1 ||
+        (i === roleGroupA.length - 1 && !isEmpty(roleGroupB))
+      ) {
+        roleList += ',';
+      }
+    });
+
+    return roleList;
+  };
+
+  const getRolesTitle = () => {
+    let roleTitle = '';
+    const assignedRoles = roles.filter((role) => role.setTo);
+    const unassignedRoles = roles.filter((role) => !role.setTo);
+
+    if (!isEmpty(assignedRoles)) {
+      roleTitle += `assigned${getFormmatedRoleList(
+        assignedRoles,
+        unassignedRoles,
+      )}`;
+    }
+
+    if (isEmpty(assignedRoles) && !isEmpty(unassignedRoles)) {
+      roleTitle += `removed${getFormmatedRoleList(
+        unassignedRoles,
+        assignedRoles,
+      )}`;
+    } else if (!isEmpty(unassignedRoles)) {
+      roleTitle += ` and removed the${getFormmatedRoleList(
+        unassignedRoles,
+        assignedRoles,
+      )}`;
+    }
+
+    roleTitle += roles.length > 1 ? ' permissions' : ' permission';
+
+    return roleTitle;
+  };
```

Comment:
The role title is not generated correctly. It always shows as the permission being set in the Root domain, even though it was set in a sub-domain.

Here's an example: I've given permissions to user "b" in domain "Second"

![Screenshot from 2021-02-05 15-52-16](https://user-images.githubusercontent.com/1193222/107045036-bce4cd00-67cd-11eb-8f8b-307264e918b5.png)

But the title still shows that permissions were added in domain "Root"

Note that the sidebar widget displays the correct domain

Reply:
Also, I think the permissions should be displayed in lower case when showing them in the titles:
- both the action page title
- and the even copy